### PR TITLE
fix: greedy variable can be any value

### PR DESCRIPTION
**Pull request by philasmar**
*Description of changes:*
* Previously, I hard coded the greedy variable as {proxy+}. In reality, the greedy variable can have any value and not "proxy" explicitly. This PR updates that behavior.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Tools/LambdaTestTool-v2/tests/Amazon.Lambda.TestTool.UnitTests/Services/ApiGatewayRouteConfigServiceTests.cs [hunk 1]
```diff
@@ -169,7 +171,7 @@ public void CatchAllRouteConfig()
             {
                 LambdaResourceName = "F1",
                 HttpMethod = "ANY",
-                Path = "/{proxy+}"
```
Comment (gcbeattyAWS) [inline Tools/LambdaTestTool-v2/tests/Amazon.Lambda.TestTool.UnitTests/Services/ApiGatewayRouteConfigServiceTests.cs:172]: can you add a test case to see what happens with `{proxy+}+}` (is this even valid?)

File: Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Services/ApiGatewayRouteConfigService.cs [hunk 2]
```diff
@@ -142,15 +142,20 @@ private bool IsRouteConfigValid(ApiGatewayRouteConfig routeConfig)
             return false;
         }
 
-        var occurrences = routeConfig.Path.Split("{proxy+}").Length - 1;
```
Comment (gcbeattyAWS) [inline Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Services/ApiGatewayRouteConfigService.cs:145]: for validating the path i think you may be able to actually just use the regex that aws uses 

```
^(\{[\w.:-]+\+?\}|[a-zA-Z0-9.:_-]+)$
```

image

would that work?

**Review by normj — APPROVED** [review]
Approving assuming you address @gcbeattyAWS comments.

File: Tools/LambdaTestTool-v2/tests/Amazon.Lambda.TestTool.UnitTests/Services/ApiGatewayRouteConfigServiceTests.cs (hunk 1 above)
Comment (philasmar) [reply Tools/LambdaTestTool-v2/tests/Amazon.Lambda.TestTool.UnitTests/Services/ApiGatewayRouteConfigServiceTests.cs:172]: Done

File: Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Services/ApiGatewayRouteConfigService.cs (hunk 2 above)
Comment (philasmar) [reply Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Services/ApiGatewayRouteConfigService.cs:145]: Done

**Review by gcbeattyAWS — APPROVED** [review]

**Pull request merged**